vertx-web-client/src/test/java/io/vertx/ext/web/client/WebClientSessionOauth2Test.java

```diff
@@ -492,4 +501,58 @@ public void testWithAuthenticationWithoutSessionExpiredWithLeeway() throws Excep
 
     awaitLatch(latchClient2);
   }
+
+  @Test
+  public void tokenInvalidatedByProvider() throws Exception {
+    final CountDownLatch latch = new CountDownLatch(1);
+    final AtomicBoolean retry = new AtomicBoolean();
+
+    server = vertx.createHttpServer().requestHandler(req -> {
+      if (req.method() == HttpMethod.POST && "/oauth/token".equals(req.path()) && !retry.get()) {
+        assertEquals("Basic Y2xpZW50LWlkOmNsaWVudC1zZWNyZXQ=", req.getHeader("Authorization"));
+        req.response().putHeader("Content-Type", "application/json").end(loggedOutFixture.encode());
```

Comment:
forgot to add retry.set(true)

Reply:
Yes it was a bit strange so I flipped the flag after sending the 401, but it makes more sense here.